plugins/modules/grafana_teams.py

```diff
@@ -0,0 +1,387 @@
+        version = response.get("version")
+        major, minor, rev = version.split(".")
+        return {"major": int(major), "minor": int(minor), "rev": int(rev)}
+
+    def create_team(self, name, email):
+        url = "/api/teams"
+        team = dict(email=email, name=name)
+        response = self._send_request(url, data=team, headers=self.headers, method="POST")
+        return response
+
+    def get_team(self, name):
+        url = "/api/teams/search?name={team}".format(team=name)
+        response = self._send_request(url, headers=self.headers, method="GET")
+        if not response.get("totalCount") <= 1:
+            raise AssertionError("Expected 1 team, got %d" % response["totalCount"])
+
+        if len(response.get("teams")) == 0:
+            return None
+        return response.get("teams")[0]
+
+    def update_team(self, team_id, name, email):
+        url = "/api/teams/{team_id}".format(team_id=team_id)
+        team = dict(email=email, name=name)
+        response = self._send_request(url, data=team, headers=self.headers, method="PUT")
+        return response
+
+    def delete_team(self, team_id):
+        url = "/api/teams/{team_id}".format(team_id=team_id)
+        response = self._send_request(url, headers=self.headers, method="DELETE")
+        return response
+
+    def get_team_members(self, team_id):
+        url = "/api/teams/{team_id}/members".format(team_id=team_id)
+        response = self._send_request(url, headers=self.headers, method="GET")
+        members = [item.get("email") for item in response]
+        return members
+
+    def add_team_member(self, team_id, email):
+        url = "/api/teams/{team_id}/members".format(team_id=team_id)
+        data = {"userId": self.get_user_id_from_mail(email)}
+        self._send_request(url, data=data, headers=self.headers, method="POST")
+
+    def delete_team_member(self, team_id, email):
+        user_id = self.get_user_id_from_mail(email)
+        url = "/api/teams/{team_id}/members/{user_id}".format(team_id=team_id, user_id=user_id)
+        self._send_request(url, headers=self.headers, method="DELETE")
+
+    def get_user_id_from_mail(self, email):
+        url = "/api/users/lookup?loginOrEmail={email}".format(email=email)
+        user = self._send_request(url, headers=self.headers, method="GET")
+        if user is None:
+            self._module.fail_json(failed=True, msg="User '%s' does not exists" % email)
+        return user.get("id")
+
+
+def setup_module_object():
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+        required_together=[['url_username', 'url_password']],
+        mutually_exclusive=[['url_username', 'grafana_api_key']],
+    )
+    return module
+
+
+argument_spec = url_argument_spec()
+# remove unnecessary arguments
+del argument_spec['force']
+del argument_spec['force_basic_auth']
+del argument_spec['http_agent']
+
+argument_spec.update(
```

Comment:
Note for the future, we could define common Grafana args and a `plugin/docs_fragment` for those common options.

Reply:
yep !

Reply:
how does the docs_fragment works ?